### ISSUE TITLE
useing a none Partition Key to querying dynamodb

### ISSUE THREAD

**Issue by PatrickMinett** [stefanbund/311#22]
I started getting an error when I tried to use the none Partition Key to specific the column. I was wondering is it possible to querying are database using something other then a partition key. ( I ended up changing my table to make it work).

**Comment by stefanbund** [comment]
It looks like you dictated this message using voice. What do you mean by none partition key?

From Stefan Bund

**Comment by PatrickMinett** [comment]
I will try to clarify, In dynamodb we are given two types of keys, the Partition key and the sort key those along with attributes we define will basically dictate what type of data that is going to have in our table ( ie in my case I have the table: Armory, I have WeaponId(partition key) and type(sort key) and other attributes like FiringRate, Model, ammunitionType. 

In are code we are given KeyCoditionExpression which if I am understanding correctly determines which attribute we are going to apply are search criteria too or so I though. In my attempts to get my database working, I found that I could only use my Partition key (ie WeaponId) to quarry my database any attempts to say refer to the attribute ammunitionType or any other None partition key would result in error code saying I needed to use the partitioned. 

my question is what if I wanted to let my user Search by a different attribute say I wanted to be able to let my user search weapons by ammunitionType? Am I stuck having to plan ahead  or is there a way I can get my code to quarry another attribute that is not Partition key ( WeaponId)

hope that helps, thanks.
